ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -579,7 +567,7 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) throws Inte
     } catch (Throwable t) {
       LOG.error("Caught an exception in the main loop of compactor worker " + workerName, t);
     } finally {
-      commitTxnIfSet(compactorTxnId);
+      compactionTxn.commit();
```

Comment:
Maybe commitOrAbort? Or close? This is not always commit if I understand correctly

Reply:
If compaction failed, it should be aborted before this. CompactionTxn#commit will only commit the txn if it's still open.